src/ray/common/ray_object.h

```diff
@@ -45,6 +45,23 @@ class RayObject {
     Init(data, metadata, nested_refs, copy_data);
   }
 
+  RayObject(const std::shared_ptr<Buffer> &metadata,
```

Comment:
Document this function. Especially point out that this constructor will "create a ray object instance whose data will be generated by the data factory"?